toolchain/parse/handle_var.cpp

```diff
@@ -16,6 +16,9 @@ static auto HandleVar(Context& context, State finish_state,
   // The finished variable declaration will start at the `var` or `returned`.
   context.PushState(state, finish_state);
 
+  // TODO: is there a cleaner way to give VarAfterPattern access to the `var`
+  // token?
+  state.token = *(context.position() - 1);
```

Comment:
We could change the token used for the `VarAs*` states to be the `var` token rather than the first token after it. (It used to be the token after because that was the start of the bracketed subtree, but we don't use the `token` field for that any more.)